File: frontend/src/types/inventory.ts
/** Inventory Optimization accelerator — domain model.
 *  Modeled on the LEAP Intelligence reference app (MRO spare-parts inventory).
 *  Mock-only for now; the same interfaces will later be served from Databricks gold.
 */

export type Region = 'Europe' | 'NORAM' | 'South Africa' | 'India' | 'ANZ' | 'LATAM';

export type RecommendationType =
  | 'reduce_stock'
  | 'dispose'
  | 'increase_stock'
  | 'keep_unchanged'
  | 'emergency_action';

export type WorkflowStatus =
  | 'pending'
  | 'eng_approved'
  | 'maint_approved'
  | 'finance_approved'
  | 'implemented'
  | 'rejected';

export type XYZ = 'X' | 'Y' | 'Z'; // demand variability
export type FSN = 'Fast' | 'Slow' | 'Non-moving'; // movement
export type VED = 'Vital' | 'Essential' | 'Desirable'; // criticality
export type RiskLevel = 'low' | 'medium' | 'high';

export interface Plant {
  id: string; // e.g. IN-LAL-01
  name: string;
  region: Region;
  country: string;
  material_count: number;
  inventory_value: number;
  service_level: number; // 0–100
}

export interface Material {
  id: string; // 8-digit code
  description: string; // "Bearings – Grundfos I935"
  category: string;
  supplier: string;
  plant_id: string;
  region: Region;
  country: string;
  xyz: XYZ;
  fsn: FSN;
  ved: VED;
  criticality_score: number; // 0–100
  coverage_days: number;
  on_hand_qty: number;
  unit_cost: number;
  current_stock_value: number;
  avg_monthly_demand: number;
}

export interface Recommendation {
  id: string;
  material_id: string;
  material_desc: string;
  category: string;
  plant_id: string;
  region: Region;
  country: string;
  type: RecommendationType;
  confidence: number; // 0–100
  current_stock_value: number;
  recommended_value: number;
  savings_potential: number;
  cash_release: number;
  risk: RiskLevel;
  ai_reasoning: string;
  coverage_days: number;
  fsn: FSN;
  ved: VED;
  criticality_score: number;
  workflow_status: WorkflowStatus;
}

/* ── UI labels / tones ─────────────────────────────────────────── */

export const REC_TYPE_LABEL: Record<RecommendationType, string> = {
  reduce_stock: 'Reduce Stock',
  dispose: 'Dispose',
  increase_stock: 'Increase Stock',
  keep_unchanged: 'Keep Unchanged',
  emergency_action: 'Emergency Action',
};

export const WORKFLOW_LABEL: Record<WorkflowStatus, string> = {
  pending: 'Pending',
  eng_approved: 'Eng. Approved',
  maint_approved: 'Maintenance Approved',
  finance_approved: 'Finance Approved',
  implemented: 'Implemented',
  rejected: 'Rejected',
};

File: backend/app/schemas/inventory.py
"""Pydantic schemas — mirror the frontend `types/inventory.ts` (snake_case) so
the UI consumes the API with no shape changes."""
from __future__ import annotations

from pydantic import BaseModel


class Plant(BaseModel):
    id: str
    name: str
    region: str
    country: str
    material_count: int
    inventory_value: float
    service_level: float

    model_config = {"from_attributes": True}


class Material(BaseModel):
    id: str
    description: str
    category: str
    supplier: str
    plant_id: str
    region: str
    country: str
    xyz: str
    fsn: str
    ved: str
    criticality_score: int
    coverage_days: int
    on_hand_qty: int
    unit_cost: float
    current_stock_value: float
    avg_monthly_demand: float

    model_config = {"from_attributes": True}


class Recommendation(BaseModel):
    id: str
    material_id: str
    material_desc: str
    category: str
    plant_id: str
    region: str
    country: str
    type: str
    confidence: int
    current_stock_value: float
    recommended_value: float
    savings_potential: float
    cash_release: float
    risk: str
    ai_reasoning: str
    coverage_days: int
    fsn: str
    ved: str
    criticality_score: int
    workflow_status: str

    model_config = {"from_attributes": True}


class MaterialPage(BaseModel):
    items: list[Material]
    total: int
    page: int
    page_size: int


class Kpis(BaseModel):
    """The 8 overview tiles."""
    total_inventory_value: float
    savings_potential: float
    service_level: float
    working_capital_released: float
    inventory_turns_yoy: float
    days_inventory_outstanding: float
    obsolete_stock_value: float
    stockouts: int
